avocado/utils/pci.py

```diff
@@ -173,14 +172,17 @@ def get_slot_from_sysfs(full_pci_address):
 
     :param full_pci_address: Full PCI address including domain (0000:03:00.0)
 
-    :return: slot of PCI address from sysfs.
+    :return: Removed port related details using re, only returns till
+    physical slot of the adapter.
     """
     if not os.path.isfile('/sys/bus/pci/devices/%s/devspec' % full_pci_address):
         return
-    devspec = read_file("/sys/bus/pci/devices/%s/devspec" % full_pci_address)
+    devspec = genio.read_file("/sys/bus/pci/devices/%s/devspec"
+                              % full_pci_address)
     if not os.path.isfile("/proc/device-tree/%s/ibm,loc-code" % devspec):
         return
-    slot = read_file("/proc/device-tree/%s/ibm,loc-code" % devspec)
+    slot = genio.read_file("/proc/device-tree/%s/ibm,loc-code" % devspec)
+    slot = re.match(r'((\w+)[.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)', slot).group()
```

Comment:
Is this guarantee to match? If not it might generate `AttributeError`... (it matches on my machine, I'm just curious why is it needed as the result is equal to the read answer:

```
>>> a="U78CB.001.WZS0CNY-P1-C32"
>>> re.match(r'((\w+)[.])+(\w+)-P(\d+)-C(\d+)|Slot(\d+)', a).group() == a
True
```